x/yarpctest/request_unary.go

```diff
@@ -113,13 +103,36 @@ func GRPCRequest(options ...api.RequestOption) api.Action {
 		require.NoError(t, out.Start())
 		defer func() { assert.NoError(t, out.Stop()) }()
 
+		sendRequestAndValidateResp(t, out, opts)
+	})
+}
+
+func sendRequestAndValidateResp(t testing.TB, out transport.UnaryOutbound, opts api.RequestOpts) {
+	f := func(i int) bool {
 		resp, cancel, err := sendRequest(out, opts.GiveRequest, opts.GiveTimeout)
 		defer cancel()
-		validateError(t, err, opts.WantError)
-		if opts.WantError == nil {
-			validateResponse(t, resp, opts.WantResponse)
+
+		if i == opts.AttemptCount-1 {
+			validateError(t, err, opts.WantError)
+			if opts.WantError == nil {
+				validateResponse(t, resp, opts.WantResponse)
+			}
+			return true
 		}
-	})
+
+		if err != nil || matchResponse(resp, opts.WantResponse) != nil {
+			return false
+		}
+
+		return true
+	}
+
+	for i := 0; i < opts.AttemptCount; i++ {
```

Comment:
Hmm, maybe I'm missing something (since tests are passing), but if we don't specify an `AttemptCount` doesn't this mean that we won't issue any requests? Not specifying should default to issuing one request.

Maybe `AttemptsCount` should be renamed to `additionalAttempts` and this condition should be `i < opts.AttemptCount + 1`?

In any case, we should probably do a validation check on the AttemptCount specified before running this loop.

Reply:
Sigh. I was wrong. Alex is right. RetryCount has the better zero value and the +1 logic you had was better that my suggestion.

Reply:
oh well, reverted. Yeah, I prefer `RetryCount` at the first place, but since it caused confusion so I thought I'd give it a different try.